sycl/include/CL/sycl/detail/pi.h

```diff
@@ -180,8 +136,9 @@ typedef enum : pi_uint64 {
   PI_DEVICE_TYPE_ALL = 0xFFFFFFFF, ///< All devices available in the PI plugin.
   PI_DEVICE_TYPE_CPU = (1 << 1),   ///< A PI device that is the host processor.
   PI_DEVICE_TYPE_GPU = (1 << 2),   ///< A PI device that is a GPU.
-  PI_DEVICE_TYPE_ACC = (1 << 3)    ///< A PI device that is a
+  PI_DEVICE_TYPE_ACC = (1 << 3),   ///< A PI device that is a
                                    ///< dedicated accelerator.
+  PI_DEVICE_TYPE_CUSTOM = (1 << 4) ///< A PI device that is a custom device.
```

Comment:
why is this added? where is it used?

Reply:
I don't think we use it anywhere, but it was added for the sake of completing the flags. sycl/tools/get_device_count_by_type.cpp is using the CL variant, so if we need to detach that from OpenCL too, then it would be ready. If you prefer we leave it out for now, I don't mind.